### PR TITLE
Support `spark.sql.parquet.int96RebaseModeInWrite=LEGACY` [databricks]

### PR DESCRIPTION
**Pull request by ttnghia**
This adds support for `LEGACY` mode in `spark.sql.parquet.int96RebaseModeInWrite`, which allows writing files containing ancient times before `1582-10-15` with rebasing from Proleptic Gregorian calendar times to Julian calendar times.

Closes:
 * https://github.com/NVIDIA/spark-rapids/issues/9037

### CODE REVIEW

**Comment by ttnghia** [comment]
build

**Comment by ttnghia** [comment]
build

**Comment by ttnghia** [comment]
build

File: integration_tests/src/main/python/parquet_write_test.py [hunk 1]
```diff
@@ -470,41 +454,17 @@ def generate_map_with_empty_validity(spark, path):
             lambda spark, path: spark.read.parquet(path),
             data_path)
 
-@pytest.mark.parametrize('ts_write_data_gen', [('INT96', TimestampGen()),
-                                               ('TIMESTAMP_MICROS', TimestampGen(start=datetime(1, 1, 1, tzinfo=timezone.utc), end=datetime(1582, 1, 1, tzinfo=timezone.utc))),
```
Comment (revans2) [inline integration_tests/src/main/python/parquet_write_test.py:474]: The timestamps here are setup to match the range that we know rebase is needed for. Are we sure that we are getting good enough coverage from just `test_datetime_roundtrip_with_legacy_rebase`?

**Review by revans2 — DISMISSED** [review]
I see the tests are updated and the code says that we now support rebase for in96 writes. But I don't see anywhere that the code was updated for it. I am assuming that the existing code just covered it and we are now enabling it after testing.

File: integration_tests/src/main/python/parquet_write_test.py (hunk 1 above)
Comment (ttnghia) [reply integration_tests/src/main/python/parquet_write_test.py:474]: Good point. I'll change the range test.

Comment (ttnghia) [reply integration_tests/src/main/python/parquet_write_test.py:474]: Done.

**Comment by ttnghia** [comment]
build

**Comment by ttnghia** [comment]
> I see the tests are updated and the code says that we now support rebase for in96 writes. But I don't see anywhere that the code was updated for it. I am assuming that the existing code just covered it and we are now enabling it after testing.

Right, the existing code already handles the rebase computation. Now we just enable the corresponding code path and update tests.

**Comment by ttnghia** [comment]
build

File: integration_tests/src/main/python/hive_write_test.py [hunk 2]
```diff
@@ -85,8 +85,6 @@ def do_write(spark, table_name):
 @pytest.mark.skipif(not is_hive_available(), reason="Hive is missing")
 @pytest.mark.parametrize("gens", [_basic_gens], ids=idfn)
 @pytest.mark.parametrize("storage_with_confs", [
-    ("PARQUET", {"spark.sql.legacy.parquet.datetimeRebaseModeInWrite": "LEGACY",
-                 "spark.sql.legacy.parquet.int96RebaseModeInWrite": "LEGACY"}),
```
Comment (revans2) [inline integration_tests/src/main/python/hive_write_test.py:89]: Why drop these?

Comment (ttnghia) [reply integration_tests/src/main/python/hive_write_test.py:89]: This is a fallback test. We now have full support for `LEGACY` in write thus we don't fallback anymore.

Comment (ttnghia) [reply integration_tests/src/main/python/hive_write_test.py:89]: This is similar: https://github.com/NVIDIA/spark-rapids/pull/9658/files#diff-33b5f03f5e0d5aa2b19d7666f2db31cb30b3c8d9b67271017ce960cc6ff40c78L789

**Review by revans2 — APPROVED** [review]

**Pull request merged**